Automattic/wp-calypso
# Help Center: move to React Router

### Changes proposed in this Pull Request

* This is a huge PR but really there is no easier way, I think. It's huge because moving the logic to the help-center meant the entry point is in a TypeScript and this forced me to type a lot of the help-center dependency tree. I supposed I could have typed these in a separate PR, though.
* This PR moves the code we added in `inline-help` to support the help-center into the help center and away from Calypso. But it doesn't free the help-center from Calypso entirely, hence the modifications in Calypso.
* It disables the contextual links tests because they're too flaky and need their own PR to find a better strategy. This change seems out of scope, but I had to move contextual-help (and it's tests) from help-center to `data-stores` because they created a circular dependency (`help-center <-> Calypso's inline-help <-> help-center`), and the breakage was breaking this PR's build, so I disabled them until I create a PR for them.
* It patches some types for Calypso imports to avoid typing entire Calypso.

### Testing instructions

Since this is a foundational PR, everything must be tested again. 

#### Testing Live chat

1. In your main user (you probably have access to live chat). Open the help center.
2. Click “Still need help”.
3. If chat is available, start a conversation, all should work.
4. Disallow 3rd party cookies, are repeat the 3 steps above. 
5. A popup should open and all should work as expected.

#### Testing Email Support
1. In your main user, open the help center.
2. Click “Still need help”.
3. Pick "Email" option.
4. Send a tick indicating that it's a test ticket and can be closed.
5. You should be redirected to the success screen.
6. Click back in the success screen, you should go back to the HC's first page.

#### Testing Forum Support
1. Using a free user open the help center.
2. Click “Still need help”.
3. You should be redirected to the forums form without any blinks.
4. Post a topic. 
5. You should be redirected to the success screen, and you should get the topic URL.
6. Click back in the success screen, you should go back to the HC's first page.

#### Testing Directly
1. Make sure you have 3rd party cookies allowed.
2. Apply D81418-code to get Directly eligibility for every user.
3. Using a free user open the help center.
4. Click “Still need help”.
7. You should be redirected to the Directly form without any flashes or blinks.
8. Send a question, the Directly UI should open.
9. Disable 3rd party cookies and refresh.
10. Reopen the help center, click “Still need help”.
11. You should be redirected to the forums. Even if you're eligible.

#### Sibyl search
1. In the contact form, searching for stuff should give you suggestions under the form. These suggestions should be clickable.

#### General search
1. Search for stuff in the help center. All should work.
2. Articles should be readable. 

#### Translation screenshot

<img width="496" alt="image" src="https://user-images.githubusercontent.com/17054134/171156839-15e2a28a-e014-47bc-9933-9ae68be0068b.png">
